app/src/ereferrals/java/org/eyeseetea/malariacare/data/sync/importer/strategies/PullControllerStrategy.java

```diff
@@ -73,12 +76,20 @@ public void onPullDataComplete(final IPullController.Callback callback, boolean
                         orgUnitDataSource.getUserOrgUnit(
                                 credentialsLocalDataSource.getOrganisationCredentials());
                 org.eyeseetea.malariacare.domain.entity.Program program = orgUnit.getProgram();
-                programLocalDataSource.saveUserProgramId(program);
+                saveProgram(programLocalDataSource, program);
             } catch (Exception e) {
                 e.printStackTrace();
                 callback.onError(e);
             }
             mPullController.convertData(callback);
         }
     }
+
+    private void saveProgram(IProgramRepository programLocalDataSource, Program program) {
+        if(!programLocalDataSource.checkLastDownloadedProgramMedia(program.getCode())){
```

Comment:
The problem here is... On logout the program associate to the user is removed. But the media only need be removed if the new user program is different than the last user program(removed on logout).
I think saving the last program uid is the most easy way.
I can add a DB field saving the program uid for each media file(always the same for each file)... Do you think is it better than save the last programuid in the preferences?